pkg/environment/api/api.go

```diff
@@ -73,22 +110,332 @@ func NewEnvironmentService(
+			zap.Error(err),
+			zap.Any("type", req.Type),
+			zap.String("code", req.Code),
+			zap.String("redirect_url", req.RedirectUrl),
+		)
+		dt, err := auth.StatusInternal.WithDetails(&errdetails.LocalizedMessage{
+			Locale:  localizer.GetLocale(),
+			Message: localizer.MustLocalize(locale.InternalServerError),
+		})
+		if err != nil {
+			return nil, auth.StatusInternal.Err()
+		}
+		return nil, dt.Err()
+	}
+	userInfo, err := authenticator.Exchange(ctx, req.Code, req.RedirectUrl)
+	if err != nil {
+		s.logger.Error("Failed to exchange",
+			zap.Error(err),
+			zap.Any("type", req.Type),
+			zap.String("code", req.Code),
+			zap.String("redirect_url", req.RedirectUrl),
+		)
+		dt, err := auth.StatusInternal.WithDetails(&errdetails.LocalizedMessage{
+			Locale:  localizer.GetLocale(),
+			Message: localizer.MustLocalize(locale.InternalServerError),
+		})
+		if err != nil {
+			return nil, auth.StatusInternal.Err()
+		}
+		return nil, dt.Err()
+	}
+	if !userInfo.VerifiedEmail {
+		s.logger.Error("Email is not verified",
+			zap.String("email", userInfo.Email),
+			zap.Any("type", req.Type),
+			zap.String("code", req.Code),
+			zap.String("redirect_url", req.RedirectUrl),
+		)
+		dt, err := auth.StatusEmailNotVerified.WithDetails(&errdetails.LocalizedMessage{
+			Locale:  localizer.GetLocale(),
+			Message: localizer.MustLocalize(locale.Account),
+		})
+		if err != nil {
+			return nil, auth.StatusEmailNotVerified.Err()
+		}
+		return nil, dt.Err()
+	}
+
+	existedInSystem, err := s.checkEmailExistedInSystem(ctx, userInfo.Email, localizer)
+	if err != nil {
+		return nil, err
+	}
+	if existedInSystem {
+		s.logger.Error("Email already exists in the system",
+			zap.String("email", userInfo.Email),
+			zap.Any("type", req.Type),
+			zap.String("code", req.Code),
+			zap.String("redirect_url", req.RedirectUrl),
+		)
+		dt, err := statusUserAlreadyInOrganization.WithDetails(&errdetails.LocalizedMessage{
+			Locale:  localizer.GetLocale(),
+			Message: localizer.MustLocalize(locale.Organization),
+		})
+		if err != nil {
+			return nil, auth.StatusInternal.Err()
+		}
+		return nil, dt.Err()
+	}
+
+	demoToken, err := s.generateDemoToken(ctx, userInfo.Email, localizer)
+	if err != nil {
+		s.logger.Error("Failed to generate demoToken",
+			zap.Error(err),
+			zap.Any("type", req.Type),
+			zap.String("code", req.Code),
+			zap.String("redirect_url", req.RedirectUrl),
+		)
+		return nil, err
+	}
+	return &environmentproto.ExchangeDemoTokenResponse{
+		DemoCreationToken: demoToken,
+	}, nil
+}
+
+func (s *EnvironmentService) checkEmailExistedInSystem(
+	ctx context.Context,
+	email string,
+	localizer locale.Localizer,
+) (bool, error) {
+	organizations, _, _, err := s.orgStorage.ListOrganizations(ctx, &mysql.ListOptions{
+		Limit:  mysql.QueryNoLimit,
+		Offset: mysql.QueryNoOffset,
+		Filters: []*mysql.FilterV2{
+			{
+				Column:   "organization.archived",
+				Operator: mysql.OperatorEqual,
+				Value:    false,
+			},
+		},
```

Comment:
I commented [here](https://github.com/bucketeer-io/bucketeer/pull/2004#discussion_r2209689759) that if the organization is archived in the future, the user can create another demo. That's why I suggested using the `GetAccountV2`, but checking the [query](https://github.com/bucketeer-io/bucketeer/blob/main/pkg/account/storage/v2/sql/account_v2/select_accounts_with_organization.sql#L33) it does not filter by archived, so the `GetMyOrganizationsByEmail` was correct.
Can you revert to using it?